src/main/java/org/sefglobal/scholarx/repository/MenteeRepository.java

```diff
@@ -36,4 +40,16 @@
             nativeQuery = true
     )
     void deleteByMentorProgramId(long id);
+
+    @Modifying
+    @Query(
+            value = "UPDATE " +
+                    "mentee " +
+                    "SET state = 'REJECTED' " +
+                    "WHERE profile_id = :profileId " +
+                    "AND program_id = :programId " +
+                    "AND mentor_id != :mentorId",
+            nativeQuery = true
+    )
+    void rejectAllByProgramIdAndProfileIdExceptMentorId(long programId, long profileId, long mentorId);
```

Comment:
```suggestion
    void rejectAllByProgramIdAndProfileIdAndMentorIdNot(long programId, long profileId, long mentorId);
```
According to the JPA ref guide